client/memory.cpp

```diff
@@ -30,6 +30,14 @@ using namespace RemoteCL;
 using namespace RemoteCL::Client;
 
 
+struct MappedPtr {
+	void* ptr;
+	size_t offset;
+	size_t size;
+};
```

Comment:
The related memory object ID should be tracked as well. We need to do a clRetain and clReleaseMemoryObject call since we're not propagating the call to the server (which would do the reference counting).